modules/core/src/lib/attribute-transition-utils.js

```diff
@@ -25,6 +26,54 @@ export function normalizeTransitionSettings(userSettings, layerSettings = {}) {
   return Object.assign({}, DEFAULT_TRANSITION_SETTINGS, layerSettings, userSettings);
 }
 
+// NOTE: NOT COPYING OVER OFFSET OR STRIDE HERE BECAUSE:
+// (1) WE DON'T SUPPORT INTERLEAVED BUFFERS FOR TRANSITIONS
+// (2) BUFFERS WITH OFFSETS ALWAYS CONTAIN VALUES OF THE SAME SIZE
+// (3) THE OPERATIONS IN THE SHADER ARE PER-COMPONENT (addition and scaling)
+export function getSourceBufferAttribute(gl, attribute) {
+  const {size, value, normalized, constant} = attribute;
+  // The Attribute we pass to Transform as a sourceBuffer must have {divisor: 0}
+  // so we create a copy of the attribute (with divisor=0) to use when running
+  // transform feedback
+  if (constant) {
+    // don't pass normalized here because the `value` from a normalized attribute is
+    // already normalized
+    return new BaseAttribute(gl, {constant, value, size});
+  }
+  const buffer = attribute.getBuffer();
+  return new BaseAttribute(gl, {divisor: 0, constant, buffer, size, normalized});
+}
+
+export function getAttributeTypeFromSize(size) {
+  switch (size) {
+    case 1:
+      return 'float';
+    case 2:
+      return 'vec2';
+    case 3:
+      return 'vec3';
+    case 4:
+      return 'vec4';
+    default:
+      throw new Error(`No defined attribute type for size "${size}"`);
```

Comment:
Nit: we generally avoid detailed error messages unless its source is the user.

Reply:
Hmm - what would you do instead of a detailed error message?

Reply:
`assert(false);`

Generally errors generated from an internal path is considered debug info and we avoid producing readable errors to reduce bundle size.

Reply:
I really think we should give as much detail as is necessary in all the failed `asserts()` and errors we throw. Error messages are just as much for the engineers who are contributing to Deck as they are for those who `npm install` Deck into their projects. These engineers include lots of newcomers to the project, many from other teams and other companies. I've run into several of those cryptic `assert() failed` error messages during development, and definitely burned a lot of time on some of them due to the lack of detail. 

If we think removing all these strings will result in a significant drop in bundle size, can we strip them out for the production bundle using a webpack plugin, like React does?